src/entry.js

```diff
@@ -23,172 +29,226 @@ class Entry {
+      next: entry.next,
+      v: entry.v,
+      clock: entry.clock
+    }
+
+    if (entry.key) Object.assign(e, { key: entry.key })
+    if (entry.identity) Object.assign(e, { identity: entry.identity })
+    if (entry.sig) Object.assign(e, { sig: entry.sig })
+
+    return dagNode.write(ipfs, 'dag-cbor', e, IPLD_LINKS)
+  }
+
+  /**
+   * Get the multihash of an Entry.
+   * @param {IPFS} ipfs An IPFS instance
+   * @param {Entry} entry Entry to get a multihash for
    * @returns {Promise<string>}
+   * @example
+   * const multihash = await Entry.toMultihash(ipfs, entry)
+   * console.log(multihash)
+   * // "Qm...Foo"
+   * @deprecated
    */
   static async toMultihash (ipfs, entry) {
     if (!ipfs) throw IpfsNotDefinedError()
-    const isValidEntryObject = entry => entry.id && entry.clock && entry.next && entry.payload && entry.v >= 0
-    if (!isValidEntryObject(entry)) {
-      throw new Error('Invalid object format, cannot generate entry multihash')
-    }
+    if (!Entry.isEntry(entry)) throw new Error('Invalid object format, cannot generate entry multihash')
 
     // Ensure `entry` follows the correct format
     const e = {
       hash: null,
       id: entry.id,
       payload: entry.payload,
       next: entry.next,
-      v: entry.v,
+      v: 0,
       clock: entry.clock
     }
 
     if (entry.key) Object.assign(e, { key: entry.key })
     if (entry.identity) Object.assign(e, { identity: entry.identity })
     if (entry.sig) Object.assign(e, { sig: entry.sig })
 
-    const data = Entry.toBuffer(e)
-    const object = await ipfs.object.put(data)
-    return object.toJSON().multihash
+    return dagNode.write(ipfs, 'dag-pb', e, IPLD_LINKS)
   }
 
   /**
-   * Create an Entry from a multihash
-   * @param {IPFS} [ipfs] An IPFS instance
-   * @param {string} [hash] Multihash as Base58 encoded string to create an Entry from
-   * @example
-   * const hash = await Entry.fromMultihash(ipfs, "Qm...Foo")
-   * console.log(hash)
-   * // { hash: "Qm...Foo", payload: "hello", next: [] }
+   * Create an Entry from a CID.
+   * @param {IPFS} ipfs An IPFS instance
+   * @param {string} cid The CID to create an Entry from
    * @returns {Promise<Entry>}
+   * @example
+   * const entry = await Entry.fromCID(ipfs, "zd...Foo")
+   * console.log(entry)
+   * // { cid: "Zd...Foo", payload: "hello", next: [] }
    */
-  static async fromMultihash (ipfs, hash) {
+  static async fromCID (ipfs, cid) {
     if (!ipfs) throw IpfsNotDefinedError()
-    if (!hash) throw new Error(`Invalid hash: ${hash}`)
+    if (!cid) throw new Error(`Invalid CID: ${cid}`)
 
-    const obj = await ipfs.object.get(hash, { enc: 'base58' })
+    const e = await dagNode.read(ipfs, cid, IPLD_LINKS)
 
-    const data = JSON.parse(obj.toJSON().data)
     let entry = {
-      hash: hash,
-      id: data.id,
-      payload: data.payload,
-      next: data.next,
-      v: data.v,
-      clock: new Clock(data.clock.id, data.clock.time)
+      cid,
```

Comment:
What's happening here? What's the key, what's the value?

Reply:
Both key and value are cid (this breaks backwards compatibility and should be `hash` if dagNode.read returns an entry with e.v === 0

Reply:
`cid` are both the key and the value, this is a feature of ES6 I think which is supported since node 4, see: https://node.green/#ES2015-syntax-object-literal-extensions-shorthand-properties

Reply:
Also, it replaces the old `hash` for the new fromCID/toCID.